### ISSUE TITLE
@htordeux, couple questions

### ISSUE THREAD

**Issue by blackcardi** [kirk24788/KPS#72]
Is there a way to update my KPS with your priest updates without KPS being updated? Also is it difficult to update the spells list for a class? If not, can you post guide on how to do so? I am pretty good at keeping my rotations updated myself but that's as much as I can do. Sure would like to be able to add spells to the spells list to help with this.

**Comment by htordeux** [comment]
> Is there a way to update my KPS with your priest updates without KPS being updated?
no, I've added new fcts in kps.env.heal to writemy holy rotation

> can you post guide on how to do so
hoho ^_^ you can install bbedit or textwrangler( free) and compare the KPS with an updated KPS . Right now I am playing TOS in heroic and the holy rotation works well it was just updated at https://github.com/htordeux/KPS except when an unit has a boss debuff like "Spear of Anguish" on Desolate Host . in these case I play manually on mouseover

**Comment by htordeux** [comment]
e.g if I want to cast serenity on a healing absorbtion debuff
 {spells.holyWordSerenity, 'heal.hasAbsorptionDebuff ~= nil' , kps.heal.hasAbsorptionDebuff },

in spells.lua , I add
```
kps.spells.absorptionDebuff = {}
kps.spells.absorptionDebuff.spearOfAnguish = kps.Spell.fromId(235933) -- Spear of Anguish
kps.spells.absorptionDebuff.EmbraceOfTheEclipse = kps.Spell.fromId(233263) -- Embrace of the Eclipse
```
in unit_auras.lua I add

```
--[[[
@function `<UNIT>.absorptionDebuff` - returns true if the unit has a Healing Absorption Debuff
]]--
function Unit.absorptionDebuff(self)
    for _,spell in pairs(kps.spells.absorptionDebuff) do
        if self.hasDebuff(spell) then return true end
    end
    return false
end
```
in heal.lua I add

```
local healAbsorptionDebuff = function(spell)
    for name, unit in pairs(raidStatus) do
        if unit.isHealable and unit.absorptionDebuff then return unit end
    end
    return nil
end

kps.RaidStatus.prototype.hasAbsorptionDebuff = kps.utils.cachedValue(function()
    return healAbsorptionDebuff
end)
```

**Comment by blackcardi** [comment]
Hey thanks for the reply. Unfortunately KPS is no longer compatible with the LUA unlocker I have been using so I am looking for a different combat rotation mod. The others I have tried can not compete with KPS. I can only hope that KPS will again work for me someday.

**Comment by htordeux** [comment]
an unlocker disable the protection for the protected lua fct castspellbyname.
a rotation addon ( jps, hps, PE engine) can be used with any unlockers.

**Comment by blackcardi** [comment]
Well, that's what I thought. KPS worked on Thursday, but did not work on Monday. To see if it was the unlocker not working, I loaded Bad Rotations and it worked with no errors using the same unlocker. That lead me to believe it was KPS that was conflicting somewhere. I am very puzzled. I am not a programmer and my only experience with it is editing premade JPS/KPS rotations. My ability to troubleshoot these issues is next to nil.
